tools/c7n_gcp/c7n_gcp/query.py

```diff
@@ -145,7 +145,45 @@ def get_resource_query(self):
             return {'filter': self.data.get('query')}
 
     def resources(self, query=None):
-        q = query or self.get_resource_query()
+        if hasattr(self.resource_type, 'parent'):
```

Comment:
no has attr please, extend the provider type info with a parent = None default

Reply:
Also all this logic should be in a subclass for resources with parents, ie ChildResourceManager.